etl/steps/__init__.py

```diff
@@ -372,8 +372,13 @@ def _output_dataset(self) -> catalog.Dataset:
         return catalog.Dataset(self._dest_dir.as_posix())
 
     def checksum_output(self) -> str:
-        # This cast from str to str is IMHO unnecessary but MyPy complains about this without it...
-        return cast(str, self._output_dataset.checksum())
+        # cache the output checksum to avoid unnecessarily re-computing checksums
+        # for the same datasets
+        key = str(self)
```

Comment:
I've never seen this before, but I guess it makes sense :-).

Reply:
All our steps implement `__str__` which returns their URI from dag (e.g. `data://garden/...`). Calling `str(self)` then calls implicitly `__str__`.